modules/imgproc/test/test_drawing.cpp

```diff
@@ -680,6 +680,120 @@ TEST(Drawing, fillpoly_circle)
     EXPECT_LT(diff_fp3, 1.);
 }
 
+TEST(Drawing, fillpoly_contours)
+{
+    int imgSize = 50;
+    int type = CV_8UC1;
+    int shift = 0;
+    cv::LineTypes lineType = LINE_8;
+
+    cv::Mat img(imgSize, imgSize, type);
+
+    // check that contours of fillPoly and polylines match
+    {
+        std::vector<cv::Point> polygonPoints;
+        polygonPoints.push_back(cv::Point(44, 27));
+        polygonPoints.push_back(cv::Point(7, 37));
+        polygonPoints.push_back(cv::Point(7, 19));
+        polygonPoints.push_back(cv::Point(38, 19));
+
+        std::vector<const cv::Point*> polygonPointPointers(polygonPoints.size());
+        for (size_t i = 0; i < polygonPoints.size(); i++)
+        {
+            polygonPointPointers[i] = &polygonPoints[i];
+        }
+
+        const cv::Point** data = &polygonPointPointers.front();
+        int size = (int)polygonPoints.size();
+        const int* npts = &size;
+        int ncontours = 1;
+
+        img = 0;
+        cv::fillPoly(img, data, npts, ncontours, 255, lineType, shift);
+        cv::polylines(img, data, npts, ncontours, true, 0, 1, lineType, shift);
+        {
+            cv::Mat labelImage(img.size(), CV_32S);
+            int labels = cv::connectedComponents(img, labelImage, 4);
+            EXPECT_EQ(2, labels) << "filling went over the border";
+        }
+    }
+
+    // check that line generated with fillPoly and polylines match
+    {
+        std::vector<cv::Point> polygonPoints;
+        polygonPoints.push_back(cv::Point(44, 27));
+        polygonPoints.push_back(cv::Point(38, 19));
+
+        std::vector<const cv::Point*> polygonPointPointers(polygonPoints.size());
+        for (size_t i = 0; i < polygonPoints.size(); i++)
+        {
+            polygonPointPointers[i] = &polygonPoints[i];
+        }
+
+        const cv::Point** data = &polygonPointPointers.front();
+        int size = (int)polygonPoints.size();
+        const int* npts = &size;
+        int ncontours = 1;
+
+        img = 0;
+        cv::fillPoly(img, data, npts, ncontours, 255, lineType, shift);
+        cv::polylines(img, data, npts, ncontours, true, 0, 1, lineType, shift);
+        EXPECT_EQ(0, cv::countNonZero(img)) << "lines do not match";
+
+        img = 0;
+        cv::polylines(img, data, npts, ncontours, true, 255, 1, lineType, shift);
+        cv::fillPoly(img, data, npts, ncontours, 0, lineType, shift);
+        EXPECT_EQ(0, cv::countNonZero(img)) << "lines do not match";
+    }
+}
+
+TEST(Drawing, DISABLED_fillpoly_match_lines)
```

Comment:
Why is this test disabled?

Reply:
It takes a lot of time compared to the other tests. The last time I added such a heavy test I was advised to not run it by default, see https://github.com/opencv/opencv/pull/23076#discussion_r1060193514

Reply:
I suggest reducing this test by:
1. use only corners and the middle as the first point:
    ```.cpp
    for (int x1 = 0; x1 < imgSize; x1 += imgSize / 2)  // imgSize = 49
        for (int y1 = 0; y1 < imgSize; y1 += imgSize / 2)
    ```
2. Draw on two separate images and compare them afterwards (see my other comment)

I've prepared an example: https://github.com/opencv/opencv/commit/1b1277b787438b3827208b7323652a892e486072 Does it cover all the required cases, what do you think?

Reply:
I adjusted the tests, thank you for the example. Should be a good amount of tests now.